checkov/common/checks_infra/solvers/attribute_solvers/base_attribute_solver.py

```diff
@@ -45,6 +53,14 @@ def get_operation(self, vertex: Dict[str, Any]) -> bool:
     def _get_operation(self, vertex: Dict[str, Any], attribute: Optional[str]) -> bool:
         raise NotImplementedError
 
+    def _process_node(self, data, passed_vartices, failed_vertices):
+        if not self.resource_type_pred(data, self.resource_types):
+            return
+        if self.get_operation(vertex=data):
+            passed_vartices.append(data)
+        else:
+            failed_vertices.append(data)
```

Comment:
It is a best practice to return new modified lists instead of inline modifying passed lists. This way you avoid hard to debug side effects (like append)

Reply:
I need to use inline modifying because multiple threads change the list in parallel.

Reply:
@YaaraVerner Do you rely on some kind of lock when modifying? Or rely on the GIL?

Reply:
I rely on the GIL. I read that append function for list is thread safe.